backend/src/main/java/org/umcs/appollo/converters/AnswerConverter.java

```diff
@@ -0,0 +1,35 @@
+package org.umcs.appollo.converters;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import org.umcs.appollo.model.AnswerEntity;
+import org.umcs.appollo.model.QuestionEntity;
+import org.umcs.appollo.model.api.Answer;
+
+import java.util.Optional;
+
+@Service
+public class AnswerConverter {
+    public Answer FromEntityToApi(AnswerEntity answerEntity) {
+        if (answerEntity == null)
+            return null;
+        Answer answer = new Answer();
+        answer.setId(answerEntity.getId());
+        answer.setQuestionId(answerEntity.getQuestion().getId());
+        answer.setAnswerJson(answerEntity.getAnswerJson());
+        return answer;
+    }
+
+    public AnswerEntity FromApiToEntity(Answer answer, Optional<QuestionEntity> questionId) {
```

Comment:
Do czego potrzebujemy tego `questionId`?
Wydaje mi się że konwerter powinien być "głupi" i przetłumaczyć dosłownie 1-1, a dodatek w postaci id question dodać w serwisie po tym dopiero jak zwróci go konwerter.

Reply:
Jak dla mnie nie ma to znaczenia poniewaz id i tak składa się na konwertowanie do entity więc jest częścią całego procesu.

Reply:
Jak najbardziej, tylko to jest logika bardziej biznesowa jak dla mnie i powinna być obsługiwana przez Service. A te konwertery mają robić podmiankę syntaktyczną, jak na przykład [Dozer](http://dozer.sourceforge.net/) czy inne biblioteki mapujące

Reply:
Usunąłem wszystko z id z convertera osoba robiąca service odpowiedzi będzie tam sobie wszystko ustawiała :)